source/base/timer.cc

```diff
@@ -353,8 +353,13 @@ TimerOutput::TimerOutput (MPI_Comm      mpi_communicator,
 
 TimerOutput::~TimerOutput()
 {
-  while (active_sections.size() > 0)
-    leave_subsection();
+  try
+    {
+      while (active_sections.size() > 0)
+        leave_subsection();
```

Comment:
What is it actually that the checker complains about here? That `leave_subsection()` is not marked as `nothrow`? I'm confused what exactly it is that we can't do in constructors. I know that we can't throw exceptions, but it looks like the checker marks more or less random functions we happen to call as potentially throwing exceptions. Would it be better to instead mark these functions as `nothrow`?